### PR TITLE
Support for joyent locations

### PR DESCRIPTION
**Pull request by rszabo50**
#535 - Added support for joyent locations

### CODE REVIEW

**Comment by rszabo50** [comment]
Appologies for the multiple commits, was getting errors on the unix command line and I missed a couple of entries copying from the shell to the web UI

File: saltcloud/clouds/joyent.py [hunk 1]
```diff
@@ -105,15 +122,19 @@ def get_conn():
...
     )
 
 
+
 def create(vm_):
     '''
     Create a single VM from a data dict
     '''
+    global conn
+
```
Comment (s0undt3ch) [inline saltcloud/clouds/joyent.py:134]: Do you really need to make it a global here?

**Comment by rszabo50** [comment]
Its because I do not call get_conn again in the create statement I will test and see if I can avoid it, does mean that get_conn will be called twice for create and stop

**Comment by rszabo50** [comment]
I have a copy that removes the need for global conn definition. You can either allow this pull request and I do a second pull request to remove the need fro global conn requirement, or you can cancel this pull and I will submit a new one 

Your choice.

File: saltcloud/clouds/joyent.py [hunk 2]
```diff
@@ -50,19 +51,15 @@
...
-list_nodes_select = namespaced_function(list_nodes_select, globals())
+JOYENT_LOCATIONS = {
+    'us-east-1': 'North Virginia, USA',
+    'us-west-1': 'Bay Area, California, USA',
+    'us-sw-1': 'Las Vegas, Nevada, USA',
+    'eu-ams-1': 'Amsterdam, Netherlands'
+}
+DEFAULT_LOCATION = 'us-east-1'
 
+conn = None
```
Comment (rszabo50) [inline saltcloud/clouds/joyent.py]: we can remove 62

File: saltcloud/clouds/joyent.py [hunk 3]
```diff
@@ -89,11 +103,14 @@ def get_configured_provider():
...
 
 
-def get_conn():
+def get_conn(location):
     '''
     Return a conn object for the passed VM data
     '''
     driver = get_driver(Provider.JOYENT)
+
+    log.info("Loading driver for connection to {0}".format(location))
```
Comment (rszabo50) [inline saltcloud/clouds/joyent.py]: change to log leve of above line to debug or remove entirely

File: saltcloud/clouds/joyent.py (hunk 1 above)
Comment (rszabo50) [inline saltcloud/clouds/joyent.py:134]: remove line 136

File: saltcloud/clouds/joyent.py [hunk 4]
```diff
@@ -216,6 +241,8 @@ def stop(name, call=None):
     '''
     Stop a node
     '''
+    global conn
```
Comment (rszabo50) [inline saltcloud/clouds/joyent.py]: remove line 244

**Comment by s0undt3ch** [comment]
Feel free to do the updates and push. That new commit will appear here.

**Comment by s0undt3ch** [comment]
`get_conn()` should be called whenever needed. And don't use the botocore driver as an example, it has some extra _behind the scenes_ stuff.

**Comment by techhat** [comment]
You're the awesome, @rszabo50!

**Comment by rszabo50** [comment]
Like I said I will be working with joyent over the next little while I may be adding more features as time goes on.

**Comment by techhat** [comment]
I'm excited to see them!

**Pull request merged**